.github/workflows/test.yml

```diff
@@ -1,178 +1,208 @@
-        if: env.GIT_DIFF
-      - name: Install GaiaV12
-        run: |
-          git checkout -
-          make build
-          cp ./build/gaiad ./build/gaiad12
-        if: env.GIT_DIFF
-      - name: Install Cosmovisor
-        run: |
-          go install github.com/cosmos/cosmos-sdk/cosmovisor/cmd/cosmovisor@latest
-        if: env.GIT_DIFF
-      - name: Start GaiaV11
-        run: |
-          go env GOPATH
-          ./contrib/scripts/upgrade_test_scripts/v12/run_gaia_v11.sh
-        if: env.GIT_DIFF
-      - name: Submit Upgrade Commands
-        run: |
-          ./contrib/scripts/upgrade_test_scripts/v12/run_upgrade_commands_v12.sh 15
-        if: env.GIT_DIFF
-      - name: Check for successful upgrade
-        run: |
-          ./contrib/scripts/upgrade_test_scripts/test_upgrade.sh 20 5 16 localhost
-        if: env.GIT_DIFF
+    upgrade-test:
+        runs-on: ubuntu-latest
+        timeout-minutes: 30
+        steps:
+            -   uses: actions/checkout@v3.5.2
+                with:
+                    fetch-depth: 0
+            -   uses: technote-space/get-diff-action@v6.0.1
+                with:
+                    PATTERNS: |
+                        **/**.go
+                        go.mod
+                        go.sum
+            -   uses: actions/setup-go@v4
+                with:
+                    go-version: 1.20.x
+            -   name: Get latest release tag
+                id: get_latest_release
+                run: |
+                    LATEST_RELEASE_TAG=$(curl -s "https://api.github.com/repos/${{ github.repository }}/releases/latest" | jq -r '.tag_name')
+                    echo "Latest release tag: $LATEST_RELEASE_TAG"
+                    echo "::set-output name=latest_release_tag::$LATEST_RELEASE_TAG"
+            -   name: Get pre-release tag if available
+                id: get_pre_release
+                run: |
+                    PRERELEASE_TAG=$(curl -s "https://api.github.com/repos/${{ github.repository }}/releases" | jq -r '.[] | select(.prerelease) | .tag_name' | head -n 1)
+                    echo "Pre-release tag: $PRERELEASE_TAG"
+                    echo "::set-output name=pre_release_tag::$PRERELEASE_TAG"
+                continue-on-error: true
+            -   name: Checkout old binary branch
+                run: |
+                    LATEST_RELEASE_TAG="${{ steps.get_latest_release.outputs.latest_release_tag }}"
+                    PRERELEASE_TAG="${{ steps.get_pre_release.outputs.pre_release_tag }}"
+            
+                    if [ -n "$PRERELEASE_TAG" ]; then
+                      echo "Using pre-release tag: $PRERELEASE_TAG"
+                      RELEASE_TAG="$PRERELEASE_TAG"
+                    else
+                      echo "No pre-release found, using latest release tag: $LATEST_RELEASE_TAG"
+                      RELEASE_TAG="$LATEST_RELEASE_TAG"
+                    fi
+                    
+                    git checkout "$RELEASE_TAG"
```

Comment:
only checkout and not build the old binary here. if build the old binary here, in the next step "checkout PR branch", the binary somehow is not there anymore, store the binary in $GITHUB_WORKSPACE also does not work.